src/me/eglp/gv2/util/base/guild/reminder/GuildReminder.java

```diff
@@ -0,0 +1,196 @@
+package me.eglp.gv2.util.base.guild.reminder;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import me.eglp.gv2.main.Graphite;
+import me.eglp.gv2.util.GraphiteUtil;
+import me.eglp.gv2.util.base.guild.GraphiteGuild;
+import me.eglp.gv2.util.base.guild.GraphiteGuildMessageChannel;
+import me.eglp.gv2.util.webinterface.js.JavaScriptClass;
+import me.eglp.gv2.util.webinterface.js.JavaScriptFunction;
+import me.eglp.gv2.util.webinterface.js.JavaScriptParameter;
+import me.eglp.gv2.util.webinterface.js.WebinterfaceObject;
+import me.mrletsplay.mrcore.json.JSONObject;
+
+/**
+ * This is the reminders executive class. It creates a Timer
+ * {@link ScheduledFuture} to execute, when the timer is due and send the
+ * appropriate message
+ * 
+ * @author The Arrayser
+ * @date Mon Mar 27 20:11:34 2023
+ */
+
+@JavaScriptClass(name = "GuildReminder")
+public class GuildReminder implements WebinterfaceObject {
+
+	// private static final DateTimeFormatter TIMESTAMP_FORMAT =
+	// DateTimeFormatter.ofPattern("EEE, dd LLL yyyy HH:mm z");
+
+	private GraphiteGuild guild;
+	private String id;
+
+	private String channelID;
+	private String message;
+	private ReminderRepetitionEnum repeatMs;
+	private LocalDateTime date;
+
+	private LocalDateTime latestPossibleDate;
+	private ScheduledFuture<?> finishFuture;
+
+	public GuildReminder(GraphiteGuild guild, LocalDateTime date, String message, ReminderRepetitionEnum repeatMs,
+			GraphiteGuildMessageChannel channel) {
+		this.guild = guild;
+		this.date = date;
+		this.latestPossibleDate = date;
+		this.message = message;
+		this.repeatMs = repeatMs;
+		this.channelID = channel.getID();
+		this.id = GraphiteUtil.randomShortID();
+	}
+
+	public GuildReminder(GraphiteGuild guild, String id, LocalDateTime date, LocalDateTime latestPossibleDate,
+			String message, ReminderRepetitionEnum repeatMs, GraphiteGuildMessageChannel channel) {
+		this.guild = guild;
+		this.date = date;
+		this.latestPossibleDate = latestPossibleDate;
+		this.message = message;
+		this.repeatMs = repeatMs;
+		this.channelID = channel.getID();
+		this.id = id;
+	}
+
+	public GuildReminder(GraphiteGuild guild, String id, String channelID, String message,
+			ReminderRepetitionEnum repeatMs, LocalDateTime date, LocalDateTime latestPossibleDate) {
+		this.guild = guild;
+		this.id = id;
+		this.channelID = channelID;
+		this.message = message;
+		this.repeatMs = repeatMs;
+		this.date = date;
+		this.latestPossibleDate = latestPossibleDate;
+	}
+
+	public String getMessage() {
+		return message;
+	}
+
+	public GraphiteGuild getGuild() {
+		return guild;
+	}
+
+	public String getChannelID() {
+		return channelID;
+	}
+
+	public String getId() {
```

Comment:
Minor issue, but please rename the method to `getID()` to be consistent with other `getID` methods in the bot

Reply:
not very camel-case-y... for the sake of camel-case-iness, ids should be written with the first letter uppercase, and the following ones lowercase, unless the following letters start a new word (words in the english language are at least 2 letters/characters long)